### DISCUSSION TITLE
How can I make a KeyArray hashable under the new system?

### DISCUSSION THREAD

**Discussion by NeilGirdhar** [jax-ml/jax#17407] [Q&A] [answered]
I have some code:
```python
        def hashable(x: Any) -> Hashable:
            if is_dataclass(x):
                return astuple(x)
            if isinstance(x, KeyArray):
                return tuple(x.unsafe_raw_array().tolist())
            ...
```
Is there a way to make this work under Jax's new `KeyArray` that acts like an ordinary array?

**Comment by jakevdp** [comment, marked as answer]
Rather than `isinstance(x, KeyArray)`, you can use `issubdtype(x.dtype, dtypes.prng_key)`. See https://github.com/google/jax/pull/17297 for some discussion.

**Reply by jakevdp** [reply]
Also, use `jax.random.key_data` rather than `unsafe_raw_array`

**Reply by NeilGirdhar** [reply]
Awesome, thanks for the quick reply as usual!
